support-frontend/assets/pages/supporter-plus-landing/twoStepPages/threeTierLanding.tsx

```diff
@@ -362,14 +391,33 @@ export function ThreeTierLanding(): JSX.Element {
 		};
 	};
 
-	const generateTierCheckoutLink = (cardTier: 1 | 2 | 3) => {
+	const getTsAndCsBaseObject = (
+		cardTier: 1 | 2 | 3,
+		contributionTypeKeyOverride?: 'annual' | 'monthly',
+	): TsAndCsProps => {
+		const cardContent = getCardContentBaseObject(
+			cardTier,
+			contributionTypeKeyOverride,
+		);
+		return {
+			title: cardContent.title,
+			planCost: cardContent.planCost,
+		};
+	};
+
+	const generateTierCheckoutLink = (cardTier: 1 | 2 | 3, promo?: Promotion) => {
```

Comment:
This method has so many conditionals, I wonder if we should just scrap it and generate the link on each card.

Reply:
Perhaps we refactor in a second PR?